plutus-pab/src/Cardano/Protocol/Socket/Client.hs

```diff
@@ -31,19 +32,26 @@ import           Cardano.Protocol.Socket.Type
 import           Ledger                                              (Block, Slot (..), Tx (..))
 
 -- | Client handler, used to communicate with the client thread.
-data ClientHandler = ClientHandler {
-        chInputQueue :: TQueue Tx,
-        chSocketPath :: FilePath,
-        chThreadId   :: ThreadId
+data ClientHandler = ClientHandler
+    { chInputQueue  :: TQueue Tx
+    , chSocketPath  :: FilePath
+    , chThreadId    :: ThreadId
+    , chCurrentSlot :: MVar Slot
```

Comment:
Shouldn't this be a `TVar`? I saw that `getCurrentSlot` uses `takeMVar`, so it leaves it empty until a new slot has started. (Also I can't see where `chCurrentSlot` is written to)